pkg/vault/vault_interface.go

```diff
@@ -0,0 +1,321 @@
+		msg := "Error deleting policy " + policyName + ":" + err.Error()
+		return errors.New(msg)
+	}
+
+	return nil
+}
+
+// DeleteSecret deletes a secret
+func (c *Connection) DeleteSecret(vaultPath string) error {
+	logicalClient := c.Client.Logical()
+	if logicalClient == nil {
+		msg := "No logical client received when deleting credentials from vault"
+		return errors.New(msg)
+	}
+	_, err := logicalClient.Delete(vaultPath)
+	if err != nil {
+		msg := "Error deleting credentials from vault for " + vaultPath + ":" + err.Error()
+		return errors.New(msg)
+	}
+	return nil
+}
+
+// GetSecret returns the stored secret as json
+func (c *Connection) GetSecret(vaultPath string) (string, error) {
+	logicalClient := c.Client.Logical()
+	if logicalClient == nil {
+		msg := "No logical client received when retrieving credentials from vault"
+		return "", errors.New(msg)
+	}
+
+	data, err := logicalClient.Read(vaultPath)
+	if err != nil {
+		msg := "Error reading credentials from vault for " + vaultPath + ":" + err.Error()
+		return "", errors.New(msg)
+	}
+
+	if data == nil || data.Data == nil {
+		msg := "No data received for credentials from vault for " + vaultPath
+		return "", errors.New(msg)
+	}
+
+	b, jsonErr := json.Marshal(data.Data)
+	if jsonErr != nil {
+		msg := "Error marshaling credentials to json for " + vaultPath + ":" + jsonErr.Error()
+		return "", errors.New(msg)
+	}
+
+	return string(b), nil
+}
+
+// AddSecret adds a secret to vault
+func (c *Connection) AddSecret(path string, credentials map[string]interface{}) error {
+	logicalClient := c.Client.Logical()
+	if logicalClient == nil {
+		msg := "No logical client received when adding secrets to vault"
+		return errors.New(msg)
+	}
+
+	_, err := logicalClient.Write(path, credentials)
+	if err != nil {
+		msg := "Error adding credentials to vault to " + path + ":" + err.Error()
+		return errors.New(msg)
+	}
+	return nil
+}
+
+// AddSecretFromStruct constructs a vault secret from the given structure
+func (c *Connection) AddSecretFromStruct(path string, creds interface{}) error {
+	jsonStr, err := json.Marshal(creds)
+	if err != nil {
+		return err
+	}
+	credentialsMap := make(map[string]interface{})
+	if err := json.Unmarshal(jsonStr, &credentialsMap); err != nil {
+		return err
+	}
+	return c.AddSecret(path, credentialsMap)
+}
+
+// Mount mounts a key-value secret provider (kv version 1) to manage the storage of the secrets
+func (c *Connection) Mount(path string) error {
+	body := strings.NewReader(`{"type":"kv-v1"}`)
+	url := c.Address + path
+
+	req, err := http.NewRequest("POST", url, body)
+	if err != nil {
+		msg := "Error creating request to mount vault for user credentials " + url + ":" + err.Error()
+		return errors.New(msg)
+	}
+	req.Header.Set("X-Vault-Token", c.Token)
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		msg := "Error mounting vault for user credentials " + url + ":" + err.Error()
```

Comment:
same here.. user creds should not be mentioned in the error...

Reply:
fixed